geonotebook/layers.py

```diff
@@ -21,25 +105,35 @@ class GeonotebookLayer(object):
     # some kind of functionality (e.g. the annotatoin layer).
     _expose_as = None
 
-    _type = None
-
-    def __init__(self, name, remote, **kwargs):
+    def __init__(self, name, remote, data, **kwargs):
```

Comment:
In general the goal of the Layer object (and its subclasses)  is to maintain the relationship between an object that provides data access (e.g. a RasterData object)  and the parameters necessary to visualize that data (e.g.  vis_url,  vis_options etc).  In certain cases that data object will be None (e.g.  OSM base layer,  annotation layer (for now)).   I agree that passing data into VisOptions is probably bad design,  but I think it probably belongs as apart of the layer object.     

Maybe this would be more clear if we changed the GeonotebookLayer signature to ```def  __init__(self, name, remote, data, vis_url, **vis_options):```  and then internally handled situations where data or vis_url is None?